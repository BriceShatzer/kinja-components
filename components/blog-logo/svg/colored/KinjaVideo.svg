<svg xmlns="http://www.w3.org/2000/svg" width="243" height="36" aria-label="Combined Shape" viewBox="0 0 243 36">
	<path d="M12.2.8h22.1l-5 9.3c-.7 1.1-2.4 2.6-5.7 2.7-3.3 0-11.3-.2-16.2 0-5 .1-7.2 2.8-7.2 2.8H14c2.2 0 4.2.3 5.8 2.7 2.2 3.6 9.6 17 9.6 17h-10a6 6 0 0 1-5.7-3.4L6.5 20a9 9 0 0 0-4.3-3.7c-1.3-.4-2.2-.5-2.2-.5v-.2L7 4.2C7.9 2.3 9.3.8 12.2.7zM80.4 10v11.6c0 3.2-2 4.7-4.8 4.7-1.3 0-2.4-.4-3.4-1.3l1-2.1a3 3 0 0 0 2.2 1c1.4 0 2.3-.9 2.3-2.4V10h2.7zm70.4-.3c4 0 6.4 3.6 6.4 8.3 0 4.7-2.4 8.3-6.4 8.3-4.1 0-6.5-3.6-6.5-8.3 0-4.7 2.4-8.3 6.5-8.3zm-48.4.3l3.5 12.8 3.5-12.8h2.9l-4.8 16h-3.2l-4.8-16h2.9zm-59 0v7.7l5-7.7h3.1l-5.2 7.6 5.6 8.4h-3.1l-4.2-6.6-1.2 1.8V26h-2.6V10h2.6zm12.9 0v16h-2.7V10h2.7zm130.5 0c2.7 0 4.1 1.8 4.1 4 0 2-1 3.4-2.5 3.7 1.6.3 2.8 2 2.8 4 0 2.5-1.4 4.3-4.2 4.3h-6.3V10zM62 10l5.8 10.5V10h2.7v16H68l-6-10.8V26h-2.7V10H62zm163 0c2.7 0 4.7 2 4.7 5 0 2.7-1.6 4.1-3 4.5l3 6.5h-2.9l-2.7-6h-2v6h-2.7V10zm-12 0l4.8 16H215l-.9-3.2h-5.3L208 26h-2.8l4.8-16h3.1zM89.5 10l4.8 16h-2.9l-.8-3.2h-5.3l-1 3.2h-2.8l4.8-16h3.2zm109.7 0c2.8 0 4.7 2 4.7 5 0 2.7-1.5 4.1-3 4.5L204 26h-3l-2.7-6h-2v6h-2.6V10zm-21.5 0v16H175V10h2.7zm-61.2 0v16h-2.6V10h2.6zm50.8 0v13.7h5.6V26h-8.3V10h2.7zm-43 0c4.4 0 6.8 3.4 6.8 8 0 4.7-2.4 8-6.7 8h-4.8V10zm18 0v2.3h-6v4.4h5.9V19h-6v4.7h6.2V26h-8.8V10h8.8zm91 0l3.4 6.8 3.4-6.8h2.9l-5 9.3V26h-2.6v-6.7l-5-9.3h3zm-82.5 2c-2.4 0-3.8 2.6-3.8 6 0 3.3 1.4 6 3.8 6s3.7-2.7 3.7-6c0-3.4-1.3-6-3.7-6zm35.5 7h-3v4.7h3c1.4 0 2.2-.9 2.2-2.4 0-1.3-.7-2.3-2.2-2.3zm-62-6.7h-2v11.4h2c2.7 0 4.1-2.4 4.1-5.7 0-3.4-1.4-5.7-4-5.7zm-36.4.5l-2.1 7.7H90l-2-7.7zm123.6 0l-2.2 7.7h4.3l-2.1-7.7zm-12.7-.5h-2.5v5.3h2.5c1.5 0 2.4-1 2.4-2.6s-1-2.7-2.4-2.7zm25.8 0H222v5.3h2.5c1.4 0 2.3-1 2.3-2.6s-.9-2.7-2.3-2.7zm-38.4 0h-2.8v4.4h2.8c1.4 0 2-1 2-2.2 0-1.2-.6-2.2-2-2.2z"/>
</svg>
