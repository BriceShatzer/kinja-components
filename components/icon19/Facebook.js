// @flow

import * as React from 'react';

import Icon19 from './icon19';
import type { Icon19Type } from './icon19';
import Facebook from './svg/Facebook.svg';

const FacebookIcon = (props: Icon19Type) =>
	<Icon19 {...props}>
		<Facebook/>
	</Icon19>;

export default FacebookIcon;
