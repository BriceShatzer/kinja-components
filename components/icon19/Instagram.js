// @flow

import * as React from 'react';

import Icon19 from './icon19';
import type { Icon19Type } from './icon19';
import Instagram from './svg/Instagram.svg';

const InstagramIcon = (props: Icon19Type) =>
	<Icon19 {...props}>
		<Instagram/>
	</Icon19>;

export default InstagramIcon;
