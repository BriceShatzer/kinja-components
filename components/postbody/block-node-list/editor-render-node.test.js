import editorRenderNode from './editor-render-node';
import { mount } from 'enzyme';

import {
	LinkNode,
	TextNode
} from 'postbody/InlineNode';
import DailyMotion from 'postbody/blockNodes/DailyMotion';
import Facebook from 'postbody/blockNodes/Facebook';
import Imgur from 'postbody/blockNodes/Imgur';
import Instagram from 'postbody/blockNodes/Instagram';
import KinjaVideo from 'postbody/blockNodes/KinjaVideo';
import Megaphone from 'postbody/blockNodes/Megaphone';
import Mixcloud from 'postbody/blockNodes/Mixcloud';
import Ooyala from 'postbody/blockNodes/Ooyala';
import Qzzr from 'postbody/blockNodes/Qzzr';
import SoundCloud from 'postbody/blockNodes/SoundCloud';
import SoundCloudPlaylist from 'postbody/blockNodes/SoundCloudPlaylist';
import Tumblr from 'postbody/blockNodes/Tumblr';
import Typeform from 'postbody/blockNodes/Typeform';
import TwitchStream from 'postbody/blockNodes/TwitchStream';
import TwitchVideo from 'postbody/blockNodes/TwitchVideo';
import Twitter from 'postbody/blockNodes/Twitter';
import Ustream from 'postbody/blockNodes/Ustream';
import UstreamLive from 'postbody/blockNodes/UstreamLive';
import Viddler from 'postbody/blockNodes/Viddler';
import Vimeo from 'postbody/blockNodes/Vimeo';
import Vine from 'postbody/blockNodes/Vine';
import Waywire from 'postbody/blockNodes/Waywire';
import YoutubePlaylist from 'postbody/blockNodes/YoutubePlaylist';
import YoutubeVideo from 'postbody/blockNodes/YoutubeVideo';

const caption = [
	new TextNode('ICE COLD! ('),
	new LinkNode([new TextNode('Flickr: Ben')], 'https://example.com'),
	new TextNode(')')
];

describe('editorRenderNode', () => {
	it('should render a link in place of a LinkPreview inset', () => {
		const node = { type: 'LinkPreview', url: 'url' };
		expect(mount(editorRenderNode(node, 1, {})).getDOMNode()).toMatchSnapshot();
	});
	it('should render a link in place of a LinkPreview inset with CommerceList style', () => {
		const node = { type: 'LinkPreview', url: 'url', style: 'CommerceList' };
		expect(mount(editorRenderNode(node, 1, {})).getDOMNode()).toMatchSnapshot();
	});
	it('should render a link in place of a LinkPreview inset with CommerceCondensed style', () => {
		const node = { type: 'LinkPreview', url: 'url', style: 'CommerceCondensed' };
		expect(mount(editorRenderNode(node, 1, {})).getDOMNode()).toMatchSnapshot();
	});
	it('should render a link in place of a ReplyInset', () => {
		const node = { type: 'ReplyInset', url: 'url'};
		expect(mount(editorRenderNode(node, 1, {})).getDOMNode()).toMatchSnapshot();
	});

	describe('embeds', () => {
		it('Soundcloud', () => {
			const wrapper = mount(editorRenderNode(new SoundCloud('220076771'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Soundcloud playlist', () => {
			const wrapper = mount(editorRenderNode(new SoundCloudPlaylist('40033455'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Facebook', () => {
			const wrapper = mount(editorRenderNode(new Facebook('10153915420718967'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		describe('Youtube', () => {
			const youtubeVideo_id = '1Y1ya-yF35g';
			const youtubeVideo_thumbnail = {
				id: 'bf8kw10mbuxnux7ttolf',
				format: 'jpg'
			};
			it('should render to embed with specified start time', () => {
				const wrapper = mount(editorRenderNode(
					new YoutubeVideo(
						youtubeVideo_id,
						youtubeVideo_thumbnail,
						{
							start: 424
						}
					), 0, {}
				));
				expect(wrapper.getDOMNode()).toMatchSnapshot();
			});
			['Left', 'Center', 'Right'].forEach(a => it(`should render with ${a} alignment`, () => {
				const wrapper = mount(editorRenderNode(
					new YoutubeVideo(
						youtubeVideo_id,
						youtubeVideo_thumbnail,
						{
							alignment: a
						}
					), 0, {}
				));
				expect(wrapper.find('aside')).toHaveClassName(`align--${a.toLowerCase()}`);
			}));
			it('should render to embed with the expected caption', () => {
				const wrapper = mount(editorRenderNode(
					new YoutubeVideo(
						youtubeVideo_id,
						youtubeVideo_thumbnail,
						{ caption }
					), 0, { captionsEnabled: true }
				));
				expect(wrapper.getDOMNode()).toMatchSnapshot();
			});
		});
		it('YoutubePlaylist', () => {
			const wrapper = mount(editorRenderNode(
				new YoutubePlaylist('RDndXwTPAFjSU', {
					id: 'zvwm06tma1hyehmettpd',
					format: 'jpg'
				}, {
					initialVideo: 'ndXwTPAFjSU',
					start: 0
				}), 0, { captionsEnabled: true }
			));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Tumblr', () => {
			const wrapper = mount(editorRenderNode(new Tumblr('gazs', '147955529854'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Typeform', () => {
			const wrapper = mount(editorRenderNode(new Typeform('clickhole', 'Zh9XTz'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('TwitchVideo', () => {
			const wrapper = mount(editorRenderNode(new TwitchVideo('69248712'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('TwitchStream', () => {
			const wrapper = mount(editorRenderNode(new TwitchStream('gosugamers'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Viddler', () => {
			const wrapper = mount(editorRenderNode(new Viddler('e5b3acca'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Vimeo', () => {
			const wrapper = mount(editorRenderNode(new Vimeo('159069175', {
				id: 'eydjkgrhuxwb76jhjcgl',
				format: 'jpg'
			}), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		describe('KinjaVideo', () => {
			it('no caption', () => {
				const wrapper = mount(editorRenderNode(new KinjaVideo('3279772', {
					id: 'gkoszxli4iwxrvxaomts',
					format: 'jpg'
				}, { alignment: undefined }), 0, {}));
				expect(wrapper.getDOMNode()).toMatchSnapshot();
			});
			it('with caption', () => {
				const wrapper = mount(editorRenderNode(new KinjaVideo('3279772',
					{ id: 'gkoszxli4iwxrvxaomts', format: 'jpg' },
					{ alignment: undefined, caption }
				), 0, { captionsEnabled: true }));
				expect(wrapper.getDOMNode()).toMatchSnapshot();
			});
		});
		it('DailyMotion', () => {
			const wrapper = mount(editorRenderNode(new DailyMotion('x44enl7', {
				id: 'ep4iblc3majmypvxbplf',
				format: 'jpg'
			}), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Ustream', () => {
			const wrapper = mount(editorRenderNode(new Ustream('84781321', {
				id: 'tvwixnozf8bjj19wbolp',
				format: 'jpg'
			}), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('UstreamLive', () => {
			const wrapper = mount(editorRenderNode(new UstreamLive('1041782'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Vine', () => {
			const wrapper = mount(editorRenderNode(new Vine('5WvdIEvMOap'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Twitter', () => {
			const wrapper = mount(editorRenderNode(new Twitter('739490015772839936'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Imgur', () => {
			const wrapper = mount(editorRenderNode(new Imgur('243598'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Instagram', () => {
			const wrapper = mount(editorRenderNode(new Instagram('BInFir8jHbf'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Mixcloud', () => {
			const wrapper = mount(editorRenderNode(new Mixcloud('https://www.mixcloud.com/dj-echo/echo-100/'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Qzzr', () => {
			const wrapper = mount(editorRenderNode(new Qzzr('165300', null, {
				height: 500
			}), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Megaphone', () => {
			const wrapper = mount(editorRenderNode(new Megaphone('DMO8510552889'), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Waywire', () => {
			const wrapper = mount(editorRenderNode(new Waywire('W265Z105JY8H0V1G', {
				id: 'ruplt40wtxxw5uqb0ycy',
				format: 'jpg'
			}, {
				alignment: 'Bleed'
			}), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
		it('Ooyala', () => {
			const wrapper = mount(editorRenderNode(new Ooyala('hocDU4NzE62S9bTYEuXDBGbPRYLsE2yT', {
				id: 'Thumbnail_Not_Found_p0odsy',
				format: 'png'
			}, {
				alignment: 'Bleed',
				playerId: 'e94d1153704449a897d545a2af16e53c'
			}), 0, {}));
			expect(wrapper.getDOMNode()).toMatchSnapshot();
		});
	});
});
