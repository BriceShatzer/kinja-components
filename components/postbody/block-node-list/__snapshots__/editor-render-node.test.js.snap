// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editorRenderNode embeds DailyMotion 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="jpg"
  data-thumb-id="ep4iblc3majmypvxbplf"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="dm-x44enl7"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="448"
      src="/ajax/inset/iframe?id=dm-x44enl7"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Facebook 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed non-flex-video"
    contenteditable="false"
    id="fb-10153915420718967"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="360"
      src="/ajax/inset/iframe?autosize=1&id=fb-10153915420718967"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Imgur 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed"
    contenteditable="false"
    id="imgur-243598"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      src="/ajax/inset/iframe?autosize=1&id=imgur-243598"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Instagram 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed non-flex-video centered-frame"
    contenteditable="false"
    id="instagram-BInFir8jHbf"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="676"
      src="/ajax/inset/iframe?autosize=1&id=instagram-BInFir8jHbf"
      width="598"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds KinjaVideo no caption 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="jpg"
  data-thumb-id="gkoszxli4iwxrvxaomts"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="kinjavideo-3279772"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?cloudinaryImageOverride=gkoszxli4iwxrvxaomts&cloudinaryImageOverrideFormat=jpg&id=kinjavideo-3279772&platform=editor"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds KinjaVideo with caption 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: #222;
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  box-shadow: inset 0 -2px 0 #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="jpg"
  data-thumb-id="gkoszxli4iwxrvxaomts"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="kinjavideo-3279772"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?cloudinaryImageOverride=gkoszxli4iwxrvxaomts&cloudinaryImageOverrideFormat=jpg&id=kinjavideo-3279772&platform=editor"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
  <footer
    contenteditable="true"
  >
    ICE COLD! (
    <span>
      <a
        class="c0 js_link c1"
        data-ga="[[\\"Embedded Url\\",\\"External link\\",\\"https://example.com\\",{\\"metric25\\":1}]]"
        href="https://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Flickr: Ben
      </a>
    </span>
    )
  </footer>
</aside>
`;

exports[`editorRenderNode embeds Megaphone 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed megaphone"
    contenteditable="false"
    id="megaphone-DMO8510552889"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="150"
      src="/ajax/inset/iframe?id=megaphone-DMO8510552889"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Mixcloud 1`] = `
<p>
  <iframe
    allowfullscreen=""
    class="custom"
    frameborder="0"
    height="180"
    mozallowfullscreen="mozallowfullscreen"
    scrolling="no"
    src="//www.mixcloud.com/widget/iframe/?feed=%2Fdj-echo%2Fecho-100%2F&hide_tracklist=1&hide_cover=1&embed_type=widget_standard"
    webkitallowfullscreen="webkitAllowFullScreen"
    width="100%"
  />
</p>
`;

exports[`editorRenderNode embeds Ooyala 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="png"
  data-thumb-id="Thumbnail_Not_Found_p0odsy"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="ooyala-hocDU4NzE62S9bTYEuXDBGbPRYLsE2yT/e94d1153704449a897d545a2af16e53c"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?id=ooyala-hocDU4NzE62S9bTYEuXDBGbPRYLsE2yT%2Fe94d1153704449a897d545a2af16e53c"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Qzzr 1`] = `
<p>
  <iframe
    allowfullscreen=""
    class="custom qzzr"
    frameborder="0"
    height="500"
    mozallowfullscreen="mozallowfullscreen"
    scrolling="no"
    src="//www.qzzr.com/widget/quiz/165300"
    webkitallowfullscreen="webkitAllowFullScreen"
    width="100%"
  />
</p>
`;

exports[`editorRenderNode embeds Soundcloud 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed non-flex-video soundcloud"
    contenteditable="false"
    id="soundcloud-220076771"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="166"
      src="/ajax/inset/iframe?id=soundcloud-220076771"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Soundcloud playlist 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed non-flex-video soundcloud"
    contenteditable="false"
    id="soundcloud-playlist-40033455"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?id=soundcloud-playlist-40033455"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Tumblr 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed"
    contenteditable="false"
    id="tumblr-post-147955529854-gazs"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="800"
      src="/ajax/inset/iframe?autosize=1&id=tumblr-post-147955529854-gazs"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds TwitchStream 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="twitch-stream-gosugamers"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="360"
      src="/ajax/inset/iframe?id=twitch-stream-gosugamers"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds TwitchVideo 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="twitch-video-69248712"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="360"
      src="/ajax/inset/iframe?id=twitch-video-69248712"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Twitter 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed twitter-embed"
    contenteditable="false"
    id="twitter-739490015772839936"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="159"
      src="/ajax/inset/iframe?autosize=1&id=twitter-739490015772839936"
      width="500"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Typeform 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed"
    contenteditable="false"
    id="typeform-clickhole-Zh9XTz"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="640"
      src="/ajax/inset/iframe?id=typeform-clickhole-Zh9XTz"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Ustream 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="jpg"
  data-thumb-id="tvwixnozf8bjj19wbolp"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="ustream-video-84781321"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="360"
      src="/ajax/inset/iframe?id=ustream-video-84781321"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds UstreamLive 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="ustream-channel-1041782"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="360"
      src="/ajax/inset/iframe?id=ustream-channel-1041782"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Viddler 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="viddler-e5b3acca"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="360"
      src="/ajax/inset/iframe?id=viddler-e5b3acca"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Vimeo 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="jpg"
  data-thumb-id="eydjkgrhuxwb76jhjcgl"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="vimeo-159069175"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?id=vimeo-159069175"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Vine 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay"
  contenteditable="false"
>
  <kinja-sandbox
    class="wysiwyg-embed non-flex-video"
    contenteditable="false"
    id="vine-5WvdIEvMOap"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="786"
      src="/ajax/inset/iframe?autosize=1&id=vine-5WvdIEvMOap"
      width="640"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Waywire 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="jpg"
  data-thumb-id="ruplt40wtxxw5uqb0ycy"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="waywire-W265Z105JY8H0V1G"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?id=waywire-W265Z105JY8H0V1G"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Youtube should render to embed with specified start time 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-start-time="424"
  data-thumb-format="jpg"
  data-thumb-id="bf8kw10mbuxnux7ttolf"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="youtube-video-1Y1ya-yF35g"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?id=youtube-video-1Y1ya-yF35g&start=424"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode embeds Youtube should render to embed with the expected caption 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  color: #222;
  -webkit-transition: color 0.1s ease-out;
  transition: color 0.1s ease-out;
  box-shadow: inset 0 -2px 0 #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #0A7BC2;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-thumb-format="jpg"
  data-thumb-id="bf8kw10mbuxnux7ttolf"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="youtube-video-1Y1ya-yF35g"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?id=youtube-video-1Y1ya-yF35g"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
  <footer
    contenteditable="true"
  >
    ICE COLD! (
    <span>
      <a
        class="c0 js_link c1"
        data-ga="[[\\"Embedded Url\\",\\"External link\\",\\"https://example.com\\",{\\"metric25\\":1}]]"
        href="https://example.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Flickr: Ben
      </a>
    </span>
    )
  </footer>
</aside>
`;

exports[`editorRenderNode embeds YoutubePlaylist 1`] = `
<aside
  class="embed-inset embed-inset--show-overlay align--bleed video-embed"
  contenteditable="false"
  data-data-initial-video="ndXwTPAFjSU"
  data-start-time="0"
  data-thumb-format="jpg"
  data-thumb-id="zvwm06tma1hyehmettpd"
>
  <kinja-sandbox
    class="wysiwyg-embed flex-video widescreen"
    contenteditable="false"
    id="youtube-list-ndXwTPAFjSU/RDndXwTPAFjSU"
  >
    <iframe
      allowfullscreen=""
      data-insecure="false"
      frameborder="0"
      height="450"
      src="/ajax/inset/iframe?id=youtube-list-ndXwTPAFjSU%2FRDndXwTPAFjSU"
      width="800"
      x-loaded="1"
    />
  </kinja-sandbox>
</aside>
`;

exports[`editorRenderNode should render a link in place of a LinkPreview inset 1`] = `
<p>
  <a
    href="url"
    x-inset="1"
    x-inset-variant="normal"
  >
    url
  </a>
</p>
`;

exports[`editorRenderNode should render a link in place of a LinkPreview inset with CommerceCondensed style 1`] = `
<p>
  <a
    href="url"
    x-inset="1"
    x-inset-variant="condensed"
  >
    url
  </a>
</p>
`;

exports[`editorRenderNode should render a link in place of a LinkPreview inset with CommerceList style 1`] = `
<p>
  <a
    href="url"
    x-inset="1"
    x-inset-variant="list"
  >
    url
  </a>
</p>
`;

exports[`editorRenderNode should render a link in place of a ReplyInset 1`] = `
<p>
  <a
    href="url"
    x-inset="1"
  >
    url
  </a>
</p>
`;
